### DISCUSSION TITLE
RFC - Sales Agent Template

### DISCUSSION THREAD

**Discussion by bschulzebaek** [shopware/frontends#522] [Ideas]
> ℹ️ This feature is supposed to be part of our commercial license and will be provided in a separated  repository with restricted access.

## Introduction
We are currently planning to build a “Sales Agent” / “Sales Representative” feature and are looking for your feedback. 

A Sales Agent is acting on behalf of a merchant, who promotes and sells their products to customers.

## Current Implementation
If you want to create an order on behalf of a customer, it is possible via the administration already. 

Since the Shopware administration isn’t as mobile friendly as we’d like to and the administration as a whole application is overkill for this scenario.

![image](https://github.com/shopware/frontends/assets/23453578/9cae2b8a-5cd4-46fe-8c66-86e3f92b7250)

### A new frontends template
We want to provide a different approach based on shopware frontends, targeting mobile devices like tablets and small notebooks specifically.

- Login page for the Sales Agent
- Customer listings
    - Customer detail page
        - Order overview
        - Quick order page

While still similar to the administration, the following wireframe draft is reducing the view to information relevant to the Sales Agent workflow, while not requiring them to click through multiple unrelated views or using complex filter components.

![image](https://github.com/shopware/frontends/assets/23453578/0417ce0d-72fe-40d7-9ea5-26aba494e34d)

**Note**: This is a draft, not everything seen is planned for the MVP.

### Sales Agent Account
The Sales Agent can be any administration user, who has the minimum set of required permissions using the existing ACL.

### New Composables
Since this feature technically already exists, we don't have to do any backend implementation but can utilize APIs the Shopware platform is providing.

Instead we will focus on wrapping these APIs in new composables which you can also use for your own implementation on top of our template.

### Customer Context
An integral part of the Sales Agent workflow is to place orders for your customers.

Creating an order for a customer via Shopware requires you to have the customer context at some point so that orders are created in the correct context and rules and extensions apply.

We’re opting for the `api.proxy.switch-customer` which is already being used by the administration implementation. This has multiple benefits:

- We won’t interrupt any workflows by adding (visible) authentication layers
- Customers won’t be asked to enter credentials into a foreign device

### What’s next?
As mentioned in the beginning, we are currently planning this feature and would be happy to get your feedback and take it into account.

---

### Update Feb 12, 2024

While discussing designs for this feature, we decided to use the [Meteor Component Library](https://github.com/shopware/meteor-component-library), since Shopware developers usually are familiar with these components already, and they'll make "Sales Agent" fit into our overall UX / UI strategy.

**Comment by htuscher** [comment]
What happens quite often: the sales agent is talking to an end customer, they prepare the order together, maybe the sales agent offers a discount on the shown items. But the customer doesn't want to buy immediately, but make his decision later.

The best solution would be that the end customer has a prepared cart which has a TTL specified by the sales agent. The items from the sales agent should be locked (not changed in qty, only deleted) and of course the customer can add regular items to the cart and place an order.

**Reply by florianholtgrefe** [reply]
From my point of view, it would be better in this case if the sales agent could create a quote that the customer can accept. This would fit better in existing workflows in the B2B Components already have.

**Reply by bschulzebaek** [reply]
I guess placing quotes or orders wouldn't make too much of a difference, you're calling different existing APIs with similar payloads. We'll keep that in mind, maybe a dedicated composable makes sense here 👍🏻

**Comment by htuscher** [comment]
If you have structured sales, you need some kind of assignment of sales agent to customer like region, zip, or some other rule.

**Comment by htuscher** [comment]
PWA: store product catalog locally because if the sales agent drives to a customer where he has no internet connection (yes happens in Bavaria quite often)

**Comment by sschwarz-allbuyone** [comment]
It would be beneficial if the sales representative could view the current shopping cart of the user and have the ability to add a coupon or discount for them.

**Comment by jochenmanz** [comment]
Most of our customers are not using a composable frontend today and also most Teams i am in contact with aren't using composable frontends in their day to day projects. Having a demo application gathers all composable frontend projects currently available would be a huge benefit for potential customers and developers likewise. When multiple b2b features are available as a composable frontend only thats a even bigger point for me :)

**Comment by marcelbrode** [comment]
Comment from the Review Bazaar, open for Votes or Discussions:
"Try to implement the "Login as the customer" feature from Shopware 5"

**Comment by marcelbrode** [comment]
Comment from the Review Bazaar, open for Votes or Discussions:
"Maybe you could add the possibility that if an order is created by a store owner, that this is noted with a tag or is visible in the admin that this order has not been placed directly by the customer."

Similar comment:
"Track which Sales Agent did the order"
